src/main/java/plugily/projects/buildbattle/arena/ArenaRegistry.java

```diff
@@ -68,12 +70,23 @@ public static BaseArena getArena(Player p) {
       return null;
     }
 
-    return getArena(playerArenaMap.get(p.getUniqueId()));
+    for (BaseArena arena : arenas) {
+      for (Player player : arena.getPlayers()) {
+        if (player.equals(p)) {
+          return arena;
+        }
+      }
+    }
+    return null;
+    //playerArenaMap??
```

Comment:
Well, `playerArenaMap` is called in `BaseArena`. Now, you find out this unecessary, remove from here and from BaseArena to make sure the players are in the proper game.

Reply:
Yes I see. I made a better temp bugfix which keeps his system intact at the moment. I'll have a talk with him later and probably remove playerArenaMap completely.
I'll close this one.